src/chain/shared_state.rs

```diff
@@ -275,12 +275,17 @@ impl SharedState {
         }
     }
 
-    /// Remove all entries from `out_members` whose name does not match `prefix`.
-    pub fn remove_our_members_not_matching_prefix(&mut self, prefix: &Prefix<XorName>) {
+    /// Remove items that are no longer relevant after a prefix change.
+    pub fn handle_prefix_change(&mut self, prefix: &Prefix<XorName>) {
+        // Prune our_members
         self.our_members = mem::replace(&mut self.our_members, BTreeMap::new())
             .into_iter()
             .filter(|(pub_id, _)| prefix.matches(pub_id.name()))
             .collect();
+
+        // Prune relocate_queue
+        self.relocate_queue
```

Comment:
Do we really want to do it here? for online node we do it when processing the event.
Question, what happen if a node was pushed in the relocate queue, but then removed before this could be processed, would you not have to handled node not there anyway?

Reply:
It's not strictly necessary to do it here, but it avoids the unnecessary work of voting `Relocate` and letting it accumulate for nodes that are no longer in our section. 

 > Question, what happen if a node was pushed in the relocate queue, but then removed

This is not handled currently, so we would still vote `Relocate`, let it accumulate and only then ignore it due to the node not being a member anymore.

Should I maybe implement the check inside `poll_relocation` instead, to handle both cases? What do you think?

Reply:
I'm thinking if it is not necessary because it is handled somewhere else already we should not try to prune it here. is this performance improvement? seems premature.

Reply:
This is already gone. I'm doing the check in `poll_relocation` now which covers both splits and nodes going offline.